packages/language-service/lib/utils/common.ts

```diff
@@ -11,17 +11,20 @@ export function findOverlapCodeRange(
 	let mappedStart: number | undefined;
 	let mappedEnd: number | undefined;
 
-	for (const [mapped, mapping] of map.getGeneratedOffsets(start)) {
-		if (filter(mapping.data)) {
-			mappedStart = mapped;
-			break;
+	for (const [mappedStart, mappedEnd] of map.getGeneratedStartEnd(start, end, filter)) {
```

Comment:
`filter` is added to map API, to optimize iterations and disregard mappings early on